src/Snap/Internal/Http/Server/Config.hs

```diff
@@ -130,6 +138,15 @@ instance Show ConfigLog where
     show (ConfigFileLog f) = "log to file " ++ show f
     show (ConfigIoLog _)   = "custom logging handler"
 
+------------------------------------------------------------------------------
+-- | This handler may be used (in conjunction with setErrorLogHandler) to write out error logs in a
+-- custom manner.
+type ErrorLogHandler = ByteString -> IO ByteString
```

Comment:
Yes, the error log handler should write the logs itself and return `()`. The idea is that out of the box we give you logging to file, in the usual format. If you want something else, you can override with a user-provided function and do whatever you want, and we will provide you with the same building blocks that we use (that's what I meant about exporting the function that generates these common log format messages to the user).

Let's say for example that (and this is a bad idea, but just for the sake of argument) you'd like to log the access log information into an SQL database. In this case, the solution you're providing is awkward, because we serialize to string without really needing to do that. In this case it's because the API we provide the user right now is bad --- the `ConfigLog` type should be broken into `AccessLogType` and `ErrorLogType`, and their equivalents to the `ConfigIoLog` constructor should have the correct, more specific types that we use internally in snap-server 1.0.

Ideally 0.10 and 1.0 should have the same API here, so that we introduce this feature once and we don't have to break user code in the upgrade from 0.10 to 1.0. In this case I'm totally OK with breaking users